### PR TITLE
26 mongodb connection

### PR DESCRIPTION
**Pull request by BrunoSienkiewicz**
- Added pipeline templates
- Added docker hub registry to push images to
- Added deploy step that triggers only on `main` branch
- Moved variables to Azure Devops

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,44 +1,37 @@
...
 stages:
-  - stage: Deploy
-    displayName: Deploy Application
-    condition: eq(variables['Build.SourceBranch'], 'refs/heads/main')
+  - stage: BuildAndTest
+    jobs:
+      - template: templates/go-test.yml
+        parameters:
+          appName: "file_transfer"
+          directory: "file-transfer/"
```
Comment (maksnowak) [inline azure-pipelines.yml:19]: Builds for file encryption, accounts and frontend will be added in the future, correct?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -1,44 +1,37 @@
...
-  REPO_URL: "https://github.com/maksnowak/FileSharingSystem"
-  WORK_DIR: "/home/azureuser/app"
+  - group: deploy-variables
+  - group: docker-registry-credentials
 
 stages:
-  - stage: Deploy
-    displayName: Deploy Application
-    condition: eq(variables['Build.SourceBranch'], 'refs/heads/main')
+  - stage: BuildAndTest
```
Comment (maksnowak) [inline azure-pipelines.yml:14]: If it builds the app and runs tests, it's gonna resolve #10, right?

File: azure-pipelines.yml (hunk 1 above)
Comment (BrunoSienkiewicz) [reply azure-pipelines.yml:19]: Yes they should be added by team members responsible for them. I provided overall template and it should be modified for the use cases, for example Vue.js probably needs different implementation.

File: azure-pipelines.yml (hunk 2 above)
Comment (BrunoSienkiewicz) [reply azure-pipelines.yml:14]: Yes but only for the go services, Vue.js should be also added in the future.

Comment (maksnowak) [reply azure-pipelines.yml:14]: Ok, I will assign it to myself then

**Review by maksnowak — APPROVED** [review]

**Pull request merged**